doc/develop/modules.rst

```diff
@@ -86,19 +86,50 @@ Module Repositories
 
 * Module repositories names should be explicitly set in the :file:`zephyr/module.yml` file.
 
-* Modules should use "zephyr" as the default name for the repository main
-  branch. Branches for specific purposes, for example, a module branch for
-  an LTS Zephyr version, shall have names starting with the 'zephyr\_' prefix.
+* If the module has an external (upstream) project repository:
 
-* If the module has an external (upstream) project repository, the module
-  repository should preserve the upstream repository folder structure.
+  * The module repository should preserve the upstream repository folder structure.
+
+  * Follow one of these strategies to track the upstream repository:
+
+    * The preferred strategy is that the module tracks directly the upstream main development
+      branch or maintenance branches which are pointed by the Zephyr manifest without any changes,
+      where all Zephyr specific code is hosted by the upstream project.
+      Pull requests are sent directly to the upstream repository.
+      Only in exceptional circumstances, when a bug fix is necessary faster than the upstream is
+      able to merge it, should the downstream, Zephyr module repository branch, be forked in a
+      separate Zephyr specific branch. This should only be done for the shortest period of time
+      possible and with the intention of returning to the upstream tracked branch as soon as
+      possible.
+
+    * For modules tracking too unresponsive upstream projects, we may need to have longer
+      lived Zephyr branches. Still, all changes in Zephyr specific branches should be submitted
+      for review to the upstream project. With the aim of abandoning these Zephyr downstream
+      branches as soon as possible.
+
+    * In some cases the upstream may not be willing to accept some Zephyr specific files or changes.
+      This is an undesirable situation as it forces us to maintain a permanent fork of the upstream
+      repository. In this case, two options are possible:
+
+      1. Relatively short lived, Zephyr specific branches are kept downstream with the Zephyr
```

Comment:
The problem with this approach is that then you cannot use Pull Requests to update to a newer version of the upstream, because you are creating a brand new branch and not updating an existing one.